research/attention_ocr/python/demo_inference.py

```diff
@@ -53,35 +54,40 @@ def load_images(file_pattern, batch_size, dataset_name):
   return images_actual_data
 
 
-def load_model(checkpoint, batch_size, dataset_name):
+def create_model(batch_size, dataset_name):
   width, height = get_dataset_image_size(dataset_name)
   dataset = common_flags.create_dataset(split_name=FLAGS.split_name)
   model = common_flags.create_model(
-      num_char_classes=dataset.num_char_classes,
-      seq_length=dataset.max_sequence_length,
-      num_views=dataset.num_of_views,
-      null_code=dataset.null_code,
-      charset=dataset.charset)
-  images_placeholder = tf.placeholder(tf.float32,
-                                      shape=[batch_size, height, width, 3])
-  endpoints = model.create_base(images_placeholder, labels_one_hot=None)
-  init_fn = model.create_init_fn_to_restore(checkpoint)
-  return images_placeholder, endpoints, init_fn
+    num_char_classes=dataset.num_char_classes,
+    seq_length=dataset.max_sequence_length,
+    num_views=dataset.num_of_views,
+    null_code=dataset.null_code,
+    charset=dataset.charset)
+  raw_images = tf.placeholder(tf.uint8, shape=[batch_size, height, width, 3])
+  images = tf.map_fn(data_provider.preprocess_image, raw_images,
+                     dtype=tf.float32)
+  endpoints = model.create_base(images, labels_one_hot=None)
+  return raw_images, endpoints
+
+
+def run(checkpoint, batch_size, dataset_name, image_path_pattern):
+  images_placeholder, endpoints = create_model(batch_size,
+                                               dataset_name)
+  images_data = load_images(image_path_pattern, batch_size,
+                            dataset_name)
+  session_creator = monitored_session.ChiefSessionCreator(
+    checkpoint_filename_with_path=checkpoint)
+  with monitored_session.MonitoredSession(
+      session_creator=session_creator) as sess:
+    predictions = sess.run(endpoints.predicted_text,
+                           feed_dict={images_placeholder: images_data})
+  return predictions.tolist()
 
 
 def main(_):
-  images_placeholder, endpoints, init_fn = load_model(FLAGS.checkpoint,
-                                                      FLAGS.batch_size,
-                                                      FLAGS.dataset_name)
-  images_data = load_images(FLAGS.image_path_pattern, FLAGS.batch_size,
-                            FLAGS.dataset_name)
-  with tf.Session() as sess:
-    tf.tables_initializer().run()  # required by the CharsetMapper
-    init_fn(sess)
-    predictions = sess.run(endpoints.predicted_text,
-                           feed_dict={images_placeholder: images_data})
   print("Predicted strings:")
-  for line in predictions:
+  for line in run(FLAGS.checkpoint, FLAGS.batch_size, FLAGS.dataset_name,
```

Comment:
Change to:
predictions = run(FLAGS.checkpoint, FLAGS.batch_size, FLAGS.dataset_name, FLAGS.image_path_pattern)
for line in predictions:
 print(line)

Reply:
Needed

Reply:
Done.